### ISSUE TITLE
Unique Column values

### ISSUE THREAD

**Issue by jrans** [dwyl/abase#54]
Some properties need to be unique in database like username and email. Provide ability to set those fields in config and initialise table with those constraints on the columns

**Comment by nelsonic** [comment]
@jrans we won't know the field needs to be _unique_ (_during validation_) until the DB replies with the constraint, so this is for table _creation_, right?
also, have you _decided_ on using `individual` as the key? 
https://github.com/dwyl/abase/pull/55#discussion_r81135409
why not `unique` ? (_would be good to have this decision documented somewhere..._)
understood that `Joi` already has a `unique()` method for `Array`: https://github.com/hapijs/joi/blob/v9.0.4/API.md#arrayuniquecomparator but does it apply to _string_ or _numeric_ values?

**Comment by jrans** [comment]
@nelsonic agreed `unique` is much nicer than `individual` and also realise only for array. Was just being super safe I guess if in future we handled arrays (what would unique constraint even mean then?)

Also more importantly adopting an attitude that you never add clashing methods is a good one to stop any nasty bugs occurring with [joi-creation](#52)  but I can imagine our handling of methods will get slightly more sophisticated by then.

Since we are not handling arrays right now (error if try to configure with one) we can safely use unique, I will change. But good we talked about this!
